### ISSUE TITLE
Can't use "switch" in view monitor

### ISSUE THREAD

**Issue by Zuploader** [darmuh/TerminalStuff#8]
Hello, just a small probleme but that's really annoying, you can't switch pov while in map or proview. Maybe you should add the same things that the FastSwitchPlayerViewInRadar mod do, just pressing left or right arrow to switch pov. WIth this, mod is perfect!

**Comment by darmuh** [comment]
To be honest I thought i'd fixed this issue before I released proview/overlay. Could you send a clip of this happening?

As for the arrow keys functionality i'll look into adding it for my own mod. In the meantime feel free to keep using both mods and let me know if you run into any compatibility issues

**Comment by Clark919** [comment]
I'm running into the same issue, here's a clip of it: https://i.imgur.com/KVWdA3n.mp4

**Comment by darmuh** [comment]
> I'm running into the same issue, here's a clip of it: https://i.imgur.com/KVWdA3n.mp4

if you use the map keyword rather than "view monitor" do you get the same issue?

**Comment by Clark919** [comment]
> if you use the map keyword rather than "view monitor" do you get the same issue?

Yep, it's the exact same issue. For reference, I'm testing this using [TheMostLethalCompany](https://thunderstore.io/c/lethal-company/p/TeamClark/TheMostLethalCompany/) if there's a potential mod there that's causing the issue. Thought I couldn't find any conflicts personally.

**Comment by darmuh** [comment]
> > if you use the map keyword rather than "view monitor" do you get the same issue?
> 
> Yep, it's the exact same issue. For reference, I'm testing this using [TheMostLethalCompany](https://thunderstore.io/c/lethal-company/p/TeamClark/TheMostLethalCompany/) if there's a potential mod there that's causing the issue. Thought I couldn't find any conflicts personally.

thanks for the info! I'll look into it and see what I can do. I only asked about the differing keywords because "view monitor" is the vanilla game code where as "map" hooks in from my code. Hence the possible differing situations.

**Comment by Zuploader** [comment]
Your mod and the other can't work together, i think yours use a command that break the other one, i even tried to disable all your "Cams" commands and it still don't work with the fastswtichmod, and the switch command still don't work

**Comment by darmuh** [comment]
tl;dr - Found the issue and seem to have fixed any compatibility issues.

https://i.imgur.com/psXj4lx.mp4

Doing some finetuning and further testing before pushing out a patch but when I do this mod should be compatible with the [FastSwitchPlayerViewInRadarMOD](https://github.com/kRYstall9/FastSwitchPlayerViewInRadarMOD).


Longer version:
The reason for the incompatibility looks to be because of the way the other mod grabbed the switch player event via the node object name "ViewInsideShipCam 1" and my mod hooks into keywords a different way that would eventually change the object name to something else. I've set all of my "player watching" commands to now use the "ViewInsideShipCam 1" object name when called and this looks to have fixed any issues.

**Comment by darmuh** [comment]
> tl;dr - Found the issue and seem to have fixed any compatibility issues.
> 
> https://i.imgur.com/psXj4lx.mp4
> 
> Doing some finetuning and further testing before pushing out a patch but when I do this mod should be compatible with the [FastSwitchPlayerViewInRadarMOD](https://github.com/kRYstall9/FastSwitchPlayerViewInRadarMOD).
> 
> Longer version: The reason for the incompatibility looks to be because of the way the other mod grabbed the switch player event via the node object name "ViewInsideShipCam 1" and my mod hooks into keywords a different way that would eventually change the object name to something else. I've set all of my "player watching" commands to now use the "ViewInsideShipCam 1" object name when called and this looks to have fixed any issues.

I also have it set so that if you run switch and nothing is on it will auto-default to enabling the map radar for now

**Comment by Zuploader** [comment]
Niiice waiting for the update then ! An other bug worth mentioning is that when using any proview or map command, if somebody else use the white button to switch player pov on the screen it's not synched for you in the terminal, don't know if it's intended but could be worth checking out!

**Comment by darmuh** [comment]
Sorry it took so long, wanted to try to fix a few more bugs alongside this one. Pushed patch 2.0.3 now with a fix for the main issue in this thread. Switch command should now properly switch between players without disabling any of the different views or causing compatibility issues with mods that also patch into the "ViewInsideShipCam 1" object name.

As for developing a feature similar to "FastSwitchPlayerViewInRadar" for this mod, I may down the line but right now I just want to focus on keeping this mod compatible with everyone else's. If I do add this particular feature I will make sure it's modular and configurable in the same way the other commands are.

Thank you @Zuploader @Clark919 and @PinipigCrunch01 from the other thread for the reports on this issue I really appreciate it 👍 


> Niiice waiting for the update then ! An other bug worth mentioning is that when using any proview or map command, if somebody else use the white button to switch player pov on the screen it's not synched for you in the terminal, don't know if it's intended but could be worth checking out!

As for this, I haven't experienced it myself but I'll look into it. On first glance this sounds like I'll need to add networking to the mod which I've already been looking into for the flashlight commands. Unfortunately networking mods in this game is not easy lol. While this didn't make it into the patch notes I'll add it as a known issue for the next one
